Classes/Utility/UserUtility.php

```diff
@@ -246,18 +246,22 @@ public static function getDirtyPropertiesFromUser(User $changedObject)
                 && !in_array($propertyName, $ignoreProperties)
             ) {
                 $newPropertyValue = $changedObject->{'get' . ucfirst($propertyName)}();
-                if (!is_object($oldPropertyValue) || !is_object($newPropertyValue)) {
+                if (!is_object($oldPropertyValue) && !is_object($newPropertyValue)) {
```

Comment:
We can move the condition of the next line into this condition.

Reply:
That condition should probably stay as is, otherwise the else part starting in line 267 is entered when it shouldn't be.